runtime/common/src/chain_ext.rs

```diff
@@ -0,0 +1,139 @@
+	/// There are no providers so the account cannot be created.
+	NoProviders,
+	/// There are too many consumers so the account cannot be created.
+	TooManyConsumers,
+	/// An error to do with tokens.
+	Token(PalletAssetTokenError),
+	/// An arithmetic error.
+	Arithmetic(PalletAssetArithmeticErr),
+	/// Unknown error
+	Unknown,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Encode, Decode, MaxEncodedLen)]
+pub enum PalletAssetArithmeticErr {
+	/// Underflow.
+	Underflow,
+	/// Overflow.
+	Overflow,
+	/// Division by zero.
+	DivisionByZero,
+	/// Unknown error
+	Unknown,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Encode, Decode, MaxEncodedLen)]
+pub enum PalletAssetTokenError {
+	/// Funds are unavailable.
+	NoFunds,
+	/// Account that must exist would die.
+	WouldDie,
+	/// Account cannot exist with the funds that would be given.
+	BelowMinimum,
+	/// Account cannot be created.
+	CannotCreate,
+	/// The asset in question is unknown.
+	UnknownAsset,
+	/// Funds exist but are frozen.
+	Frozen,
+	/// Operation is not supported by the asset.
+	Unsupported,
+	/// Unknown error
+	Unknown,
+}
+
+impl From<DispatchError> for ChainExtensionErr {
+	fn from(e: DispatchError) -> Self {
+		match e {
+			DispatchError::Other(_) => ChainExtensionErr::Other,
+			DispatchError::CannotLookup => ChainExtensionErr::CannotLookup,
+			DispatchError::BadOrigin => ChainExtensionErr::BadOrigin,
+			DispatchError::Module(_) => ChainExtensionErr::Module,
+			DispatchError::ConsumerRemaining => ChainExtensionErr::ConsumerRemaining,
+			DispatchError::NoProviders => ChainExtensionErr::NoProviders,
+			DispatchError::TooManyConsumers => ChainExtensionErr::TooManyConsumers,
+			DispatchError::Token(token_err) =>
+				ChainExtensionErr::Token(PalletAssetTokenError::from(token_err)),
+			DispatchError::Arithmetic(arithmetic_error) =>
+				ChainExtensionErr::Arithmetic(PalletAssetArithmeticErr::from(arithmetic_error)),
+			_ => ChainExtensionErr::Unknown,
+		}
+	}
+}
+
+impl From<ArithmeticError> for PalletAssetArithmeticErr {
+	fn from(e: ArithmeticError) -> Self {
+		match e {
+			ArithmeticError::Underflow => PalletAssetArithmeticErr::Underflow,
+			ArithmeticError::Overflow => PalletAssetArithmeticErr::Overflow,
+			ArithmeticError::DivisionByZero => PalletAssetArithmeticErr::DivisionByZero,
+		}
+	}
+}
+
+impl From<TokenError> for PalletAssetTokenError {
+	fn from(e: TokenError) -> Self {
+		match e {
+			TokenError::NoFunds => PalletAssetTokenError::NoFunds,
+			TokenError::WouldDie => PalletAssetTokenError::WouldDie,
+			TokenError::BelowMinimum => PalletAssetTokenError::BelowMinimum,
+			TokenError::CannotCreate => PalletAssetTokenError::CannotCreate,
+			TokenError::UnknownAsset => PalletAssetTokenError::UnknownAsset,
+			TokenError::Frozen => PalletAssetTokenError::Frozen,
+			TokenError::Unsupported => PalletAssetTokenError::Unsupported,
+		}
+	}
+}
+
+pub fn try_from(type_id: u8, code: [u8; 12], issuer: [u8; 32]) -> Result<CurrencyId, ()> {
```

Comment:
Can we have a better name for this? I feel like I commented this before but I can't find the comment lol.

Anyways, since this is exported and used as is, I would prefer to have a descriptive name like: `try_get_currency_id_from()`

Reply:
Done